src/core/src/op/constant.cpp

```diff
@@ -261,7 +261,8 @@ void Constant::allocate_buffer(bool memset_allocation) {
         constexpr uint8_t init_value = 0;
         m_data = std::make_shared<AlignedBuffer>(byte_size, host_alignment());
 
-        if (memset_allocation) {
+        // AlignedBuffer allocates 1 byte for empty constants, and we set it to zero
+        if (memset_allocation || m_data->size() != byte_size) {
```

Comment:
`m_data->size() != byte_size` check might be wider than the current scenario when `m_data->size() == 1` and `byte_size == 0`. Can it cause any side effects?

Reply:
yes, you are right, i missed that, e.g when we align the buffer and the allocated size will be increased, let me fix that

Reply:
fixed